src/main/scala/org/hammerlab/guacamole/commands/StructuralVariantCaller.scala

```diff
@@ -34,100 +38,165 @@ object StructuralVariant {
+      r.inferredInsertSize * sgn
     }
 
-    // Given a sorted sequence of positions (all of which are multiples of @blockSize), coalesce ranges of
-    // consecutive multiples of @blockSize into tuples representing the start and end (inclusive) of each range.
-    // For example, (0, 10, 20, 30, 70, 80) --> [(0, 30), (70, 80)]
-    def coalesceAdjacent(xs: IndexedSeq[Long], blockSize: Long): Iterator[(Long, Long)] = {
-      val rangeStarts = ArrayBuffer(0)
-      for (i <- 1 until xs.length) {
-        if (xs(i - 1) + blockSize != xs(i)) {
-          rangeStarts += i
-        }
+    // Extract median and MAD (Median Absolute Deviation) from an unordered Array.
+    def medianStats[T <% Double](xs: Array[T])(implicit ord: Ordering[T]): MedianStats = {
+      if (xs.isEmpty) {
+        return MedianStats(0.0, 0.0)
       }
-      rangeStarts += xs.length
 
-      for {
-        Seq(startIdx, nextStartIdx) <- rangeStarts.sliding(2)
-      } yield {
-        (xs(startIdx), xs(nextStartIdx - 1))
+      // Extract the median of a sorted array of numbers
+      def getMedian[T <% Double](nums: Array[T]): Double = {
+        val len = nums.length
+        if (len % 2 == 0) {
+          0.5 * (nums(len / 2 - 1) + nums(len / 2))
+        } else {
+          1.0 * nums(len / 2)
+        }
       }
+
+      val nums = xs.sorted
+      val median = getMedian(nums)
+      val residuals = nums.map(x => Math.abs(1.0 * x - median)).sorted
+      val mad = getMedian(residuals)
+
+      MedianStats(median, mad)
     }
 
-    // Returns a sequence of all blocks which overlap the read, its mate or the insert between them.
-    def matedReadBlocks(r: PairedRead[MappedRead]): Seq[Long] = {
-      if (!r.isMateMapped) {
-        Seq[Long]()
-      } else {
-        val mateStart = r.mateAlignmentProperties.map(_.start).getOrElse(r.read.start)
-        val start = Math.min(r.read.start, mateStart)
-        val roundedStart = start / BLOCK_SIZE * BLOCK_SIZE
-        val insertSize = orientedInsertSize(r).getOrElse(0)
-        roundedStart to (roundedStart + insertSize) by BLOCK_SIZE
+    case class ExceptionalReadsReturnType(
+      val readsInRange: RDD[PairedMappedRead],
+      val insertSizes: RDD[Int],
+      val insertStats: MedianStats,
+      val maxNormalInsertSize: Int,
+      val exceptionalReads: RDD[PairedMappedRead])
+
+    def getExceptionalReads(reads: RDD[PairedMappedRead]): ExceptionalReadsReturnType = {
+      // Pare down to mate pairs which aren't highly displaced & aren't inverted or translocated.
+      val readsInRange = reads.filter(r => {
+        val s = r.inferredInsertSize
+        r.read.referenceContig == r.mate.referenceContig &&
+          r.read.isPositiveStrand != r.mate.isPositiveStrand &&
+          s < MAX_INSERT_SIZE
+      })
+      readsInRange.persist()
+
+      val insertSizes = readsInRange.map(orientedInsertSize)
+      val insertStats = medianStats(insertSizes.takeSample(false, 100000))
```

Comment:
thoughts on getting the actual median / MAD? cf. [SO for RDD's median](http://stackoverflow.com/a/28160731/544236)